client/components/SignUpBox.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react";
+import gql from "graphql-tag";
+
+export const addUserMutation = gql`
+  mutation addUser($email: String!, $password: String!) {
+    addUser(email: $email, password: $password) {
+      email
+      password
+    }
+  }
+`;
+
+const SignUpBox = ({ apolloClient }) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+  return (
+    <>
+      <form className="form-signin mt-5">
+        <h1 className="mb-3">Sign Up</h1>
+        <div className="form-group">
+          <label htmlFor="email" className="float-left required">
+            Email address
+          </label>
+          <input
+            type="email"
+            className="form-control"
+            id="email"
+            placeholder="Enter your email"
+            value={email}
+            onChange={event => {
+              setEmail(event.target.value);
+            }}
+            autoFocus
+            required
+          />
+          <small
+            id="emailHelp"
+            className="form-text text-muted float-left mb-3"
+          >
+            We&apos;ll never share your email with anyone else.
+          </small>
+          <br />
+        </div>
+        <div className="form-group">
+          <label htmlFor="password" className="float-left required">
+            Password
+          </label>
+          <input
+            type="password"
+            className="form-control"
+            id="password"
+            placeholder="Enter your password"
+            value={password}
+            onChange={event => {
+              setPassword(event.target.value);
+            }}
+            required
+          />
+          <small
+            id="emailHelp"
+            className="form-text text-muted float-left mb-3"
+          >
+            Must contain at least 6 alphanumeric characters.
+          </small>
+          <br />
+        </div>
+
+        <button
+          type="submit"
+          className="btn btn-primary float-left"
+          onClick={async event => {
+            setErrorMessage("");
+            setSuccessMessage("");
+            event.preventDefault();
+            if (password.length < 6) {
+              return setErrorMessage(
```

Comment:
Even if both length and alphanumeric checks fail, it seems to me that it only shows an error message for length check because it returns from the function here.
Is this what you expect?

Reply:
Yes. I wanted it to handle one error at a time. First, make the user aware of length issue, and then the alphanumeric problem if it exists.